src/main/java/com/achdev/onlinebookstoreapp/mapper/UserMapper.java

```diff
@@ -3,16 +3,34 @@
 import com.achdev.onlinebookstoreapp.config.MapperConfig;
 import com.achdev.onlinebookstoreapp.dto.user.UserDto;
 import com.achdev.onlinebookstoreapp.dto.user.UserRegistrationRequestDto;
+import com.achdev.onlinebookstoreapp.exception.EntityNotFoundException;
 import com.achdev.onlinebookstoreapp.mapper.password.EncodedMapping;
 import com.achdev.onlinebookstoreapp.mapper.password.PasswordEncoderMapper;
+import com.achdev.onlinebookstoreapp.model.Role;
 import com.achdev.onlinebookstoreapp.model.User;
+import com.achdev.onlinebookstoreapp.repository.role.RoleRepository;
+import java.util.Set;
+import org.mapstruct.AfterMapping;
 import org.mapstruct.Mapper;
 import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+import org.springframework.beans.factory.annotation.Autowired;
 
-@Mapper(config = MapperConfig.class, uses = PasswordEncoderMapper.class)
-public interface UserMapper {
-    UserDto toDto(User user);
+@Mapper(config = MapperConfig.class, uses = {PasswordEncoderMapper.class})
+public abstract class UserMapper {
+    @Autowired
+    protected RoleRepository roleRepository;
+
+    public abstract UserDto toDto(User user);
 
     @Mapping(source = "password", target = "password", qualifiedBy = EncodedMapping.class)
-    User toModel(UserRegistrationRequestDto requestDto);
+    public abstract User toModel(UserRegistrationRequestDto requestDto);
+
+    @AfterMapping
+    public void setDefaultRole(@MappingTarget User user) {
```

Comment:
this approach could cause some problems
this is business logic that should be in service layer
hard to testing
and others

let's replace this logic to service

Reply:
Ok